### ISSUE TITLE
Popover / Tooltip: prop "AccessibilityType" does not work correct

### ISSUE THREAD

**Issue by MarcusMorba** [uber/baseweb#5025]
## Current Behavior
The trigger element for the popover/tooltip always has the attributes <aria-haspopup="true" aria-expanded="false">, which should only happen if the default value "menu" is applied.

## Expected Behavior
The trigger element for the popover/tooltip should not have attributes <aria-haspopup="true" aria-expanded="false">, when the AccessibilityType is set to "none" or "tooltip" as described in the "Accessibility"-section here: https://baseweb.design/components/popover/

## Your Environment
Your website 
v11: https://baseweb.design/components/popover/ & https://baseweb.design/components/tooltip/
v10: https://v10.baseweb.design/components/popover/ & https://v10.baseweb.design/components/tooltip/

| Tech    | Version |
| ------- | ------- |
| Base UI | v10.x & v11.x
| React   |  v17.x
| browser |chrome latest windows 10

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by MarcusMorba** [comment]
Reference:  https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role ()

The tooltip is not considered a popup in terms of the [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) property on the owning element, which is why we used "text bubble" in the introductory definition.

Though a tooltip may appear and disappear, as its appearance is automatic and not intentionally controlled by the user, the [aria-expanded](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) role is not supported.
